### PR TITLE
Fixes for basic profiling

### PR DESCRIPTION
**Pull request by actgardner**
I tried to build the Ruby gem with basic profiling enabled (`GRPC_CONFIG=basicprof`) on OS X 10.12.5 and ran into two issues:
- setting `GRPC_TIMERS_RDTSC` doesn't compile on my system
- `GPR_TIMER_BEGIN` in `grpc_error_string` is in the wrong place, which leads to unclosed spans in the latency profile

### CODE REVIEW

File: src/core/lib/iomgr/error.cc [hunk 1]
```diff
@@ -727,10 +727,10 @@ static char *finish_kvs(kv_pairs *kvs) {
 }
 
 const char *grpc_error_string(grpc_error *err) {
-  GPR_TIMER_BEGIN("grpc_error_string", 0);
   if (err == GRPC_ERROR_NONE) return no_error_string;
   if (err == GRPC_ERROR_OOM) return oom_error_string;
   if (err == GRPC_ERROR_CANCELLED) return cancelled_error_string;
+  GPR_TIMER_BEGIN("grpc_error_string", 0);
```
Comment (ctiller) [inline src/core/lib/iomgr/error.cc]: Now that we have C++... why not move this back to the head of function and use GPR_TIMER_SCOPE (removing the GPR_TIMER_END)

File: src/core/lib/support/time_precise.cc [hunk 2]
```diff
@@ -24,8 +24,8 @@
 
 #ifdef GRPC_TIMERS_RDTSC
 #if defined(__i386__)
-static void gpr_get_cycle_counter(int64_t int *clk) {
-  int64_t int ret;
+static void gpr_get_cycle_counter(int64_t *clk) {
+  int64_t ret;
```
Comment (ctiller) [inline src/core/lib/support/time_precise.cc]: Yikes, looks good :)

**Comment by ctiller** [comment]
This is ok to test

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_InProcessCHTTP2_NoOpMutator_NoOpMutator__2M_1_0.opt.old: 1
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_InProcessCHTTP2_NoOpMutator_NoOpMutator__2M_1_0.counters.new: 1


[microbenchmarks] No significant performance differences
```

**Comment by vjpai** [comment]
Sorry, can you rebase this against the current master? Unfortunately, you sent in this PR at the exact moment that master had lost sanity, so the tests are failing sanity. A quick rebase and retest should get that back to normal. Thanks!

**Comment by thelinuxfoundation** [comment]
Thank you for your pull request. Before we can look at your contribution, we need to ensure all contributors are covered by a Contributor License Agreement.

After the following items are addressed, please respond with a new comment here, and the automated system will re-verify.

* User @alanctgardner isn't covered by a CLA. They will need to complete the form at https://identity.linuxfoundation.org/projects/cncf

_Regards,
The Linux Foundation CLA GitHub bot_

**Comment by vjpai** [comment]
@alanctgardner : Sorry, this PR stayed out for a while and our CLA changed in the meanwhile. I resolved conflicts just now but can you follow the instructions for signing the CLA. Thanks!

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by alanctgardner** [comment]
@vjpai CLA is signed, I'm not sure what's up with the failing CI

**Comment by thelinuxfoundation** [comment]
Thank you for your pull request. Before we can look at your contribution, we need to ensure all contributors are covered by a Contributor License Agreement.

After the following items are addressed, please respond with a new comment here, and the automated system will re-verify.

* One or more commits are not associated with a GitHub user. The email used in the Git commit must match a verified email address in a GitHub user profile. For more information, please see https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/

_Regards,
CLA GitHub bot_

**Pull request closed without merging**